File: maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int max = Integer.MIN_VALUE, sum = 0;
        
        for (int num : nums) {
            if (sum < 0) {
                sum = num;
            } else {
                sum += num;
            }
            
            if (sum > max) {
                max = sum;
            }
            
        }
        
        return max;
    }
}

File: maximum-subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr = float('-inf')
        
        largest = float('-inf')
        for i in range(len(nums)):
            curr = max(curr + nums[i], nums[i])
            if curr > largest:
                largest = curr
        return largest
        
        
